test/js/web/fetch/fetch-readablestream-request-body.test.ts

```diff
@@ -0,0 +1,131 @@
+import { describe, expect, test } from "bun:test";
+
+describe("fetch with Request body lifecycle", () => {
+  test("should properly handle Request with cloned body", async () => {
+    using server = Bun.serve({
+      port: 0,
+      async fetch(req) {
+        const text = await req.text();
+        return new Response(text);
+      },
+    });
+
+    const originalRequest = new Request(server.url, {
+      method: "POST",
+      body: "test data",
+    });
+
+    // clone creates a new request that might use ReadableStream internally
+    const clonedRequest = originalRequest.clone();
+
+    const response = await fetch(clonedRequest);
+    expect(await response.text()).toBe("test data");
+
+    // original should still be usable
+    const response2 = await fetch(originalRequest);
+    expect(await response2.text()).toBe("test data");
+  });
+
+  test("should handle aborted fetch with Request body", async () => {
+    using server = Bun.serve({
+      port: 0,
+      async fetch() {
+        await Bun.sleep(100);
+        return new Response("ok");
+      },
+    });
+
+    const request = new Request(server.url, {
+      method: "POST",
+      body: "test data that might become a stream",
```

Comment:
This will never reach the code path because `body` does not become a `ReadableStream` and we use a separate optimization that detects the data is buffered. So you could probably delete this test.